test/inductor/test_cpu_select_algorithm.py

```diff
@@ -523,6 +523,38 @@ def forward(self, x, other, other2):
                 0,
             )
 
+    @inductor_config.patch({"freezing": True})
+    @patches
+    @torch.no_grad
+    @parametrize("batch_size", (3, 16, 32, 49))
+    @parametrize("in_features", (4, 68, 128))  # k should be a multiple of 4
+    @parametrize("out_features", (32, 64))  # n should be a multiple of block_n
+    @parametrize("bias", (True, False))
+    def test_quantized_linear_amx(self, batch_size, in_features, out_features, bias):
+        class M(torch.nn.Module):
+            def __init__(self, bias):
+                super().__init__()
+                self.linear = torch.nn.Linear(in_features, out_features, bias)
+
+            def forward(self, x):
+                return self.linear(x)
+
+        counters.clear()
+        v = torch.randn(batch_size, in_features).to(dtype=torch.float32)
+        ref_quantized_mod = _generate_qdq_quantized_model(
+            M(bias=bias).eval(),
+            (v,),
+        )
+        atol, rtol = 1e-2, 1e-2
+        with patch.object(select_algorithm, "VERIFY", dict(atol=atol, rtol=rtol)):
+            self.common(ref_quantized_mod, (v,), atol=atol, rtol=rtol)
+        self.assertEqual(counters["inductor"]["select_algorithm_autotune"], 1)
+        vec_amx = VecAMX()
+        if vec_amx:
+            self.assertTrue(counters["inductor"]["cpp_micro_gemm_amx_counter"] > 0)
+        else:
+            self.assertEqual(counters["inductor"]["cpp_micro_gemm_amx_counter"], 0)
```

Comment:
refactor this into a util checking function?

Reply:
Sure, added a util function to check AMX usage.